### PR TITLE
Photoupload

### LINKED ISSUE #38: Picture Upload Page
- [ ] #37 
- [ ] Upload div
- [ ] input "BildName"
- [ ] input "Bildbeschreibung
- [ ] send Button

### PR DESCRIPTION
**Pull request by lauravikanis**
People might want to upload pictures for Locations themselves. Give them an Uploader!
close #38

### CODE REVIEW

File: client/src/components/Button/Button.js [hunk 1]
```diff
@@ -8,8 +8,8 @@ const Button = styled.input`
   background-color: var(--primary-color);
   border-radius: 15px;
   border: none;
-  width: 25%;
-  max-width: 150px;
+  /* width: 25%; */
+  /* max-width: 150px; */
```
Comment (boldCM) [inline client/src/components/Button/Button.js:12]: could you delete these comments?

File: client/src/components/Image/ImageAdd.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import styled from "styled-components/macro";
+import PropTypes from "prop-types";
+import { Link } from "react-router-dom";
+
+const ImageContainer = styled(Link)`
+  object-fit: scale-down;
```
Comment (boldCM) [inline client/src/components/Image/ImageAdd.js:6]: interesting! never used that one before!:)

File: lib/Images.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+// ) {
+//   const UploadImage = await collection("pictures").InsertOne(
+//     { location: locationName },
+//     { name: imageName },
+//     { description: description },
+//     { image: image }
+//   );
+
+//   return UploadImage;
+// }
```
Comment (boldCM) [inline lib/Images.js:39]: delete comments ;)

File: server.js [hunk 4]
```diff
@@ -64,14 +66,31 @@ app.get("/api/tour/:tour", async (req, res) => {
   res.sendFile(path.join(__dirname, "client/build", "index.html"));
 });
 
+//ImageRoutes
+app.get("/api/locationImages/:locationName", async (req, res) => {
+  const { locationName } = req.query;
+  try {
+    const locationValue = await getImageDataOfLocation(locationName);
+    if (!locationValue) {
+      res.status(404).send("could not find the content you are looking for");
```
Comment (boldCM) [inline server.js:75]:
```suggestion
      res.status(404).send("Could not find the content you are looking for");
```

**Review by boldCM — APPROVED** [review]
lgtm 🎉

File: server.js (hunk 4 above)
Comment (lauravikanis) [reply server.js:90]: 😄  just saw your comments now. fixed a lot of things here and there

File: client/src/api/upload.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  const imgEntry = await response.json();
+  return imgEntry;
+};
+
+export async function postImageObj(image) {
+  const response = await fetch(`/api/upload/`, {
+    method: "POST",
+    body: JSON.stringify(image),
+    headers: {
+      "Content-Type": "application/json",
```
Comment (boldCM) [inline client/src/api/upload.js:12]: not mandatory, but maybe better if you also incoorporate:
```suggestion
      "Content-Type": "application/json;charset=utf-8",
```

File: client/src/pages/Location.js [hunk 6]
```diff
@@ -77,16 +79,24 @@ const Location = () => {
               {locationByName.pic.map((locationPic) => (
                 <>
                   <Image
-                    key={locationPic.name}
-                    locationName={name}
+                    key={locationPic.link}
+                    locationname={name}
```
Comment (boldCM) [inline client/src/pages/Location.js:83]: not sure here, but wouldn't you also want locationPic.name for locationname?
Also use camelCase for locationName ;)

File: client/src/pages/PictureUpload.js [hunk 7]
```diff
@@ -0,0 +1,106 @@
+import React, { useState } from "react";
+// import { useHistory } from "react-router-dom";
+import styled from "styled-components/macro";
+
+import UploadPreview from "../components/Image/UploadPreview";
+// import Input from "../components/Input/Input";
+import { useParams } from "react-router-dom";
```
Comment (boldCM) [inline client/src/pages/PictureUpload.js:7]: delete comments ;)

File: client/src/pages/PictureUpload.js [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+          location: locationname.name,
+        }),
+        headers: { "Content-Type": "application/json" },
+      });
+      setUploadInput("");
+      setPreviewSrc("");
+    } catch (error) {
+      console.error(error);
+    }
+  };
```
Comment (boldCM) [inline client/src/pages/PictureUpload.js:81]: you should refractor this later on ;)

**Review by boldCM — APPROVED** [review]
lgtm 🎉

File: client/src/pages/Location.js (hunk 6 above)
Comment (lauravikanis) [reply client/src/pages/Location.js:83]: No, sorry locationName and locationPic.name are two different things. the first one is the location name and the second one is the name of a picture. Going to change the camelcase in locationName 👍

File: client/src/pages/PictureUpload.js (hunk 8 above)
Comment (lauravikanis) [reply client/src/pages/PictureUpload.js]: outsourced this one into image.js 👍

**Pull request merged**